dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNLoadInterfaceIIDM.cpp

```diff
@@ -0,0 +1,160 @@
+//
+// Copyright (c) 2015-2020, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+// This file is part of Dynawo, an hybrid C++/Modelica open source time domain
+// simulation tool for power systems.
+//
+
+//======================================================================
+/**
+ * @file  DYNLoadInterfaceIIDM.cpp
+ *
+ * @brief Load data interface : implementation file for IIDM structure
+ *
+ */
+//======================================================================
+
+#include "DYNLoadInterfaceIIDM.h"
+
+#include <powsybl/iidm/Load.hpp>
+
+using boost::shared_ptr;
+using powsybl::iidm::Load;
+
+namespace DYN {
+
+LoadInterfaceIIDM::~LoadInterfaceIIDM() {
+}
+
+LoadInterfaceIIDM::LoadInterfaceIIDM(Load& load) : InjectorInterfaceIIDM(load, load.getId()),
+                                                   loadIIDM_(load),
+                                                   loadPUnderV_(0.),
+                                                   v0_(0.),
+                                                   vNom_(0.) {
+  setType(ComponentInterface::LOAD);
+  stateVariables_.resize(3);
+  stateVariables_[VAR_P] = StateVariable("p", StateVariable::DOUBLE);       // P
+  stateVariables_[VAR_Q] = StateVariable("q", StateVariable::DOUBLE);       // Q
+  stateVariables_[VAR_STATE] = StateVariable("state", StateVariable::INT);  // connectionState
+}
+
+int
+LoadInterfaceIIDM::getComponentVarIndex(const std::string& varName) const {
+  int index = -1;
+  if (varName == "p")
+    index = VAR_P;
+  else if (varName == "q")
+    index = VAR_Q;
+  else if (varName == "state")
+    index = VAR_STATE;
+  return index;
+}
+
+void
+LoadInterfaceIIDM::exportStateVariablesUnitComponent() {
+  bool connected = (getValue<int>(VAR_STATE) == CLOSED);
+
+  loadIIDM_.getTerminal().setP(getValue<double>(VAR_P) * SNREF);
+  loadIIDM_.getTerminal().setQ(getValue<double>(VAR_Q) * SNREF);
+
+  if (loadIIDM_.getTerminal().isConnected()) {
+    if (connected)
+      loadIIDM_.getTerminal().connect();
+    else
+      loadIIDM_.getTerminal().disconnect();
+  }
+}
+
+void
+LoadInterfaceIIDM::importStaticParameters() {
+  staticParameters_.clear();
+  double P = InjectorInterfaceIIDM::getP();
```

Comment:
No need for InjectorInterface prefix

Reply:
same for line 79, 99 ...

Reply:
Some methods in LoadInterfaceIIDM must be instantiated since "pure virtual" from the abstract DYNLoadInterface. It is the case for GetID() setBusInterface() and others. Some are not mandatory; in that case we don't need to specify InjectorInterfaceIIDM as a prefix. We dont' even have to specify the method since the compiler finds the injector's one. But when we must provide a method, only because of "pure virtual", it is mandatory to specify the prefix. Otherwise the method loops on itself and generates a segmentation fault. Do you agree  ? I changed the file following your indication, but some prefix must stay.

Reply:
Not sure I got your point. My point is that we can implement the pure virtual methods directly in the injector and thus there would be no need to put them again the child classes

Reply:
Should I test that point ? I don't have an answer yet...

Reply:
I quickly made the tests and what I had in mind does not seem to work so let's drop this point

Reply:
I see what's happening: in case of a derived of a derived class, you may omit the intermediary method implementing the pure virtual one. It works fine (I checked). In our case LoadInterfaceIIDM is a derived class of LoadInterface and InjectorInterface, but InjectorInterface does not derive from LoadInterface itself, mayking some method mandatory I guess !